### PR TITLE
build: HttpTemplate as separate target

### PR DESCRIPTION
**Pull request by lizan**
Envoy transcoding will depends on this.

### CODE REVIEW

File: contrib/endpoints/src/api_manager/BUILD [hunk 1]
```diff
@@ -164,7 +174,7 @@ cc_test(
     ],
     linkstatic = 1,
     deps = [
-        ":api_manager",
+        ":http_template",
```
Comment (qiwzhang) [inline contrib/endpoints/src/api_manager/BUILD]: You only need http_template?   How about patch_match files?

Comment (lizan) [inline contrib/endpoints/src/api_manager/BUILD]: Right, `PathMatcher` and `PathMatcherBuilder` will be in next PR :) I'm trying to keep PRs as small as possible.

Comment (qiwzhang) [inline contrib/endpoints/src/api_manager/BUILD]: They works together, so they should be in the same PR.

Comment (lizan) [inline contrib/endpoints/src/api_manager/BUILD]: `HttpTemplate` doesn't have any dependency so it is easier to be separated, `PathMatcher` needs a lot of refactoring.

**Pull request merged**